diff --git a/similarity_search/CMakeLists.txt b/similarity_search/CMakeLists.txt
index 03a3275..440355a 100644
--- a/similarity_search/CMakeLists.txt
+++ b/similarity_search/CMakeLists.txt
@@ -36,7 +36,7 @@ endif()
 #message(FATAL_ERROR "stopping... compiler version is: ${CMAKE_CXX_COMPILER_ID} ${CXX_COMPILER_VERSION}")
 
 if(NOT NMSLIB_SIMD_FLAGS)
-    set(NMSLIB_SIMD_FLAGS "-march=native" CACHE STRING "SIMD compile flags for NMSLIB" FORCE)
+    set(NMSLIB_SIMD_FLAGS "-mcpu=native" CACHE STRING "SIMD compile flags for NMSLIB" FORCE)
 endif()
 
 set(WARN_FLAGS "-Wall -Wunreachable-code -Wcast-align") 
