--- examples/wheel/BUILD
+++ examples/wheel/BUILD
@@ -207,6 +207,7 @@
     platform = select({
         ":aarch64-apple-darwin": "macosx_11_0_arm64",
         ":aarch64-unknown-linux-gnu": "manylinux2014_aarch64",
+        ":ppc64le-unknown-linux-gnu": "manylinux2014_ppc64le",
         ":x86_64-apple-darwin": "macosx_11_0_x86_64",  # this is typically macosx_10_9_x86_64?
         ":x86_64-pc-windows-msvc": "win_amd64",
         ":x86_64-unknown-linux-gnu": "manylinux2014_x86_64",
