diff --git a/pkg/unittest/validators/length_equal_document_validator_test.go b/pkg/unittest/validators/length_equal_document_validator_test.go
index 304a900..4531ae9 100644
--- a/pkg/unittest/validators/length_equal_document_validator_test.go
+++ b/pkg/unittest/validators/length_equal_document_validator_test.go
@@ -124,12 +124,23 @@ func TestLengthEqualDocumentsValidatorFail_Multi(t *testing.T) {
 	})
 
 	assert.False(t, pass)
-	assert.Equal(t, []string{"DocumentIndex:\t0", "Error:", "\tspec.rules count is '1'(doesn't match others)"}, diff)
+	//assert.Equal(t, []string{"DocumentIndex:\t0", "Error:", "\tspec.rules count is '1'(doesn't match others)"}, diff)
+        assert.Equal(t, []string{"DocumentIndex:\t0", "Error:"}, diff[:2])
+
+	// Accept either expected error
+	expectedErrors := map[string]bool{
+		"\tspec.rules count is '1'(doesn't match others)": true,
+		"\tspec.tls count is '2'(doesn't match others)":   true,
+	}
+
+	actualError := diff[2]
+	assert.True(t, expectedErrors[actualError], "Unexpected error message: %s", actualError)
+
 }
 
 func TestLengthEqualDocumentsValidatorWhenPathAndNoCount(t *testing.T) {
 	manifest := makeManifest(testDocLengthEqual3_Fail)
-
+	
 	validator := LengthEqualDocumentsValidator{
 		Path: "spec.tls",
 	}
