diff --git a/engines/pytorch/pytorch-jni/build.gradle b/engines/pytorch/pytorch-jni/build.gradle
index 3e7371577..7007b4ea5 100644
--- a/engines/pytorch/pytorch-jni/build.gradle
+++ b/engines/pytorch/pytorch-jni/build.gradle
@@ -19,12 +19,14 @@ processResources {
     doFirst {
         String url = "https://publish.djl.ai/pytorch/${ptVersion}/jnilib/${djl_version}"
         def files = [
+                "linux-ppc64le/cpu/libdjl_torch.so",
                 "linux-x86_64/cpu/libdjl_torch.so",
                 "linux-x86_64/cpu-precxx11/libdjl_torch.so",
                 "osx-x86_64/cpu/libdjl_torch.dylib",
                 "win-x86_64/cpu/djl_torch.dll"
         ]
         if (ptVersion.startsWith("1.13.")) {
+            files.add("linux-ppc64le/cpu/libdjl_torch.so")
             files.add("linux-aarch64/cpu-precxx11/libdjl_torch.so")
             files.add("linux-x86_64/cu117/libdjl_torch.so")
             files.add("linux-x86_64/cu117-precxx11/libdjl_torch.so")
diff --git a/engines/pytorch/pytorch-native/build.gradle b/engines/pytorch/pytorch-native/build.gradle
index 62bd3e025..0ae176b8d 100644
--- a/engines/pytorch/pytorch-native/build.gradle
+++ b/engines/pytorch/pytorch-native/build.gradle
@@ -31,7 +31,7 @@ String BINARY_ROOT = "${project.buildDir}/download"
 version = VERSION + (isRelease ? "" : "-SNAPSHOT")

 def downloadBuild(String ver, String os, String flavor, Boolean isPrecxx11 = false, Boolean isAarch64 = false) {
-    def arch = isAarch64 ? "aarch64" : "x86_64"
+    def arch = isAarch64 ? "aarch64" : "ppc64le"
     exec {
         if (os == "win") {
             commandLine "${project.projectDir}/build.cmd", "${ver}", "${flavor}"
@@ -62,7 +62,7 @@ def downloadBuild(String ver, String os, String flavor, Boolean isPrecxx11 = fal
 }

 def downloadBuildAndroid(String ver) {
-    def LIST_ABI = ["armeabi-v7a", "arm64-v8a", "x86", "x86_64"]
+    def LIST_ABI = ["armeabi-v7a", "arm64-v8a", "x86", "x86_64","ppc64le"]
     LIST_ABI.each {
         abi ->
             exec {
@@ -96,6 +96,7 @@ def prepareNativeLib(String binaryRoot, String ver) {
     }

     def files = [
+            "cpu/libtorch-cxx11-abi-shared-with-deps-${ver}%2Bcpu.zip"        : "cpu/linux-ppc64le",
             "cpu/libtorch-cxx11-abi-shared-with-deps-${ver}%2Bcpu.zip"        : "cpu/linux-x86_64",
             "cpu/libtorch-macos-${ver}.zip"                                   : "cpu/osx-x86_64",
             "cpu/libtorch-win-shared-with-deps-${ver}%2Bcpu.zip"              : "cpu/win-x86_64",
@@ -271,6 +272,7 @@ task uploadS3 {

         def f = new File("${BINARY_ROOT}/files.txt")
         def uploadDirs = [
+                "${BINARY_ROOT}/cpu/linux-ppc64le/native/lib/",
                 "${BINARY_ROOT}/cpu/linux-x86_64/native/lib/",
                 "${BINARY_ROOT}/cpu/osx-aarch64/native/lib/",
                 "${BINARY_ROOT}/cpu/osx-x86_64/native/lib/",
diff --git a/extensions/tokenizers/build.gradle b/extensions/tokenizers/build.gradle
index d49146778..2aa7cc7c7 100644
--- a/extensions/tokenizers/build.gradle
+++ b/extensions/tokenizers/build.gradle
@@ -23,6 +23,7 @@ processResources {
                 "win-x86_64/libstdc%2B%2B-6.dll" : "extra",
                 "win-x86_64/tokenizers.dll"      : "${tokenizers_version}/jnilib/${djl_version}",
                 "linux-x86_64/libtokenizers.so"  : "${tokenizers_version}/jnilib/${djl_version}",
+                "linux-ppc64le/libtokenizers.so"  : "${tokenizers_version}/jnilib/${djl_version}",
                 "linux-aarch64/libtokenizers.so" : "${tokenizers_version}/jnilib/${djl_version}",
                 "osx-x86_64/libtokenizers.dylib" : "${tokenizers_version}/jnilib/${djl_version}",
                 "osx-aarch64/libtokenizers.dylib": "${tokenizers_version}/jnilib/${djl_version}",
