diff --git a/build.gradle b/build.gradle
index e9efe3f1..cc3ba917 100644
--- a/build.gradle
+++ b/build.gradle
@@ -47,8 +47,7 @@ buildscript {
         mavenLocal()
         mavenCentral()
         maven { url "https://plugins.gradle.org/m2/" }
-        maven { url "https://central.sonatype.com/repository/maven-snapshots/" }
-        maven { url "https://artifacts.opensearch.org/snapshots/lucene/" }
+        maven { url "https://ci.opensearch.org/ci/dbc/snapshots/maven/" }
         maven { url "https://build.shibboleth.net/nexus/content/groups/public" }
     }
 
@@ -422,7 +421,7 @@ publishing {
     repositories {
         maven {
             name = "Snapshots" //  optional target repository name
-            url = "https://central.sonatype.com/repository/maven-snapshots/"
+            url = "https://ci.opensearch.org/ci/dbc/snapshots/maven/"
             credentials {
                 username System.getenv("SONATYPE_USERNAME")
                 password System.getenv("SONATYPE_PASSWORD")
@@ -494,6 +493,7 @@ configurations {
 
             // For org.opensearch.plugin:transport-grpc
             force "com.google.guava:failureaccess:1.0.3"
+            exclude group: "org.lz4", module: "lz4-java"
         }
     }
 
@@ -506,8 +506,7 @@ allprojects {
         mavenLocal()
         mavenCentral()
         maven { url "https://plugins.gradle.org/m2/" }
-        maven { url "https://central.sonatype.com/repository/maven-snapshots/" }
-        maven { url "https://artifacts.opensearch.org/snapshots/lucene/" }
+        maven { url "https://ci.opensearch.org/ci/dbc/snapshots/maven/" }
         maven { url "https://build.shibboleth.net/nexus/content/repositories/releases" }
         maven { url "build.shibboleth.net/maven/releases"}
     }
@@ -711,7 +710,7 @@ dependencies {
     runtimeOnly "org.glassfish.jaxb:jaxb-runtime:${jaxb_version}"
     runtimeOnly 'com.google.j2objc:j2objc-annotations:3.1'
     compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
-    runtimeOnly 'org.lz4:lz4-java:1.8.0'
+    runtimeOnly 'org.lz4:lz4-java:1.10.2'
     runtimeOnly 'org.slf4j:slf4j-api:1.7.36'
     runtimeOnly "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}"
     runtimeOnly 'org.xerial.snappy:snappy-java:1.1.10.8'
