diff --git a/build.gradle b/build.gradle
index 0b25c6f..e27b5dd 100644
--- a/build.gradle
+++ b/build.gradle
@@ -138,7 +138,9 @@ dependencies {
     compileOnly("com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}")
 
     spark 'org.apache.spark:spark-sql-api_2.13:3.5.4'
-    spark 'org.apache.spark:spark-core_2.13:3.5.4'
+    spark ('org.apache.spark:spark-core_2.13:3.5.4') {
+    exclude group: 'org.eclipse.jetty', module: 'jetty-server'
+    }
     spark group: 'org.apache.spark', name: 'spark-common-utils_2.13', version: '3.5.4'
 
     implementation 'org.scala-lang:scala-library:2.13.9'
@@ -223,6 +225,8 @@ task addSparkJar(type: Copy) {
         // Remove the unwanted directory from jar B
         delete file("${jarBContents}/org/apache/spark/unused")
         delete file("${jarBContents}/org/sparkproject/jetty/http")
+        delete file("${jarBContents}/org/sparkproject/jetty/server")
+        delete file("${jarBContents}/META-INF/maven/org.eclipse.jetty/jetty-server")
         delete file("${jarBContents}/META-INF/maven/org.eclipse.jetty/jetty-http")
         // Re-compress jar B
         ant.zip(destfile: jarB, baseDir: jarBContents)
