diff --git a/paddle/phi/backends/cpu/cpu_info.cc b/paddle/phi/backends/cpu/cpu_info.cc
index 3c492aa97c..2a5dab0809 100644
--- a/paddle/phi/backends/cpu/cpu_info.cc
+++ b/paddle/phi/backends/cpu/cpu_info.cc
@@ -151,7 +151,8 @@ bool MayIUse(const cpu_isa_t cpu_isa) {
   } else {
 #if !defined(WITH_NV_JETSON) && !defined(PADDLE_WITH_ARM) &&  \
     !defined(PADDLE_WITH_SW) && !defined(PADDLE_WITH_MIPS) && \
-    !defined(PADDLE_WITH_LOONGARCH)
+    !defined(PADDLE_WITH_LOONGARCH) &&                         \
+    !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
     std::array<int, 4> reg;
     cpuid(reg.data(), 0);
     int nIds = reg[0];
diff --git a/paddle/phi/backends/cpu/cpu_info.h b/paddle/phi/backends/cpu/cpu_info.h
index 3fcb5538e8..06ed178c8e 100644
--- a/paddle/phi/backends/cpu/cpu_info.h
+++ b/paddle/phi/backends/cpu/cpu_info.h
@@ -42,7 +42,8 @@
 #else
 #if !defined(WITH_NV_JETSON) && !defined(PADDLE_WITH_ARM) &&  \
     !defined(PADDLE_WITH_SW) && !defined(PADDLE_WITH_MIPS) && \
-    !defined(PADDLE_WITH_LOONGARCH)
+    !defined(PADDLE_WITH_LOONGARCH) &&                         \
+    !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
 #include <cpuid.h>
 inline void cpuid(int reg[4], int x) {
   __cpuid_count(x, 0, reg[0], reg[1], reg[2], reg[3]);
diff --git a/paddle/phi/core/platform/denormal.cc b/paddle/phi/core/platform/denormal.cc
index 93c55e84be..f9c34c869a 100644
--- a/paddle/phi/core/platform/denormal.cc
+++ b/paddle/phi/core/platform/denormal.cc
@@ -30,7 +30,8 @@
 
 #if !defined(GCC_WITHOUT_INTRINSICS) && !defined(PADDLE_WITH_ARM) && \
     !defined(PADDLE_WITH_SW) && !defined(PADDLE_WITH_MIPS) &&        \
-    !defined(_WIN32) && !defined(PADDLE_WITH_LOONGARCH)
+    !defined(_WIN32) && !defined(PADDLE_WITH_LOONGARCH) &&           \
+    !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
 #define DENORM_USE_INTRINSICS
 #endif
 
diff --git a/paddle/phi/kernels/funcs/search_compute.h b/paddle/phi/kernels/funcs/search_compute.h
index b51f10cce9..cf921d3361 100644
--- a/paddle/phi/kernels/funcs/search_compute.h
+++ b/paddle/phi/kernels/funcs/search_compute.h
@@ -15,7 +15,8 @@
 #pragma once
 
 #if !defined(PADDLE_WITH_ARM) && !defined(PADDLE_WITH_SW) && \
-    !defined(PADDLE_WITH_MIPS) && !defined(PADDLE_WITH_LOONGARCH)
+    !defined(PADDLE_WITH_MIPS) && !defined(PADDLE_WITH_LOONGARCH) && \
+    !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
 #include <immintrin.h>
 #endif
 #include <cfloat>
@@ -101,7 +102,8 @@ void call_gemm_batched(const Context& ctx,
 }
 
 #if !defined(PADDLE_WITH_ARM) && !defined(PADDLE_WITH_SW) && \
-    !defined(PADDLE_WITH_MIPS) && !defined(PADDLE_WITH_LOONGARCH)
+    !defined(PADDLE_WITH_MIPS) && !defined(PADDLE_WITH_LOONGARCH) && \
+    !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
 
 #define __m256x __m256
 
@@ -142,7 +144,8 @@ inline void axpy(const T* x, T* y, size_t len, const T alpha) {
                       _mm256_mul_px(mm_alpha, _mm256_load_px(x + jjj))));
   }
 #elif defined(PADDLE_WITH_ARM) || defined(PADDLE_WITH_SW) || \
-    defined(PADDLE_WITH_MIPS) || defined(PADDLE_WITH_LOONGARCH)
+    defined(PADDLE_WITH_MIPS) || defined(PADDLE_WITH_LOONGARCH) || \
+    defined(__powerpc__) || defined(__ppc__) || defined(__PPC__)
   PADDLE_THROW(common::errors::Unimplemented("axpy is not supported"));
 #else
   lll = len & ~SSE_CUT_LEN_MASK;
@@ -172,7 +175,8 @@ inline void axpy_noadd(const T* x, T* y, size_t len, const T alpha) {
     _mm256_store_px(y + jjj, _mm256_mul_px(mm_alpha, _mm256_load_px(x + jjj)));
   }
 #elif defined(PADDLE_WITH_ARM) || defined(PADDLE_WITH_SW) || \
-    defined(PADDLE_WITH_MIPS) || defined(PADDLE_WITH_LOONGARCH)
+    defined(PADDLE_WITH_MIPS) || defined(PADDLE_WITH_LOONGARCH) || \
+    defined(__powerpc__) || defined(__ppc__) || defined(__PPC__)
   PADDLE_THROW(common::errors::Unimplemented("axpy_noadd is not supported"));
 #else
   lll = len & ~SSE_CUT_LEN_MASK;
diff --git a/paddle/phi/kernels/funcs/softmax_impl.h b/paddle/phi/kernels/funcs/softmax_impl.h
index 8f6b0fdd32..b27ed69f36 100644
--- a/paddle/phi/kernels/funcs/softmax_impl.h
+++ b/paddle/phi/kernels/funcs/softmax_impl.h
@@ -237,10 +237,10 @@ class SoftmaxFunctor<DeviceContext, T, enable_if_CPU<DeviceContext>> {
                   const int axis_dim,
                   const phi::DenseTensor* X,
                   phi::DenseTensor* Y) {
+#if defined(__AVX__) && !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
     const auto& in_dims = X->dims();
     constexpr int kBatchDim = 0;
     constexpr int kClassDim = 1;
-
     const int num_classes = in_dims[kClassDim];
     const int batch_size = in_dims[kBatchDim];
     const int num_remain = num_classes / axis_dim;
@@ -267,7 +267,9 @@ class SoftmaxFunctor<DeviceContext, T, enable_if_CPU<DeviceContext>> {
         in_data += num_classes;
         out_data += num_classes;
       }
-    } else {
+    } else
+#endif
+    {
       SoftmaxEigen<DeviceContext, T>()(context, axis_dim, X, Y);
     }
   }
@@ -394,13 +396,13 @@ class SoftmaxGradFunctor<DeviceContext, T, enable_if_CPU<DeviceContext>> {
                   const phi::DenseTensor* y,
                   const phi::DenseTensor* y_grad,
                   phi::DenseTensor* x_grad) {
+#if defined(__AVX__) && !defined(__powerpc__) && !defined(__ppc__) && !defined(__PPC__)
     const auto& out_dims = y->dims();
     constexpr int kBatchDim = 0;
     constexpr int kClassDim = 1;
     const int num_classes = out_dims[kClassDim];
     const int batch_size = out_dims[kBatchDim];
     const int num_remain = num_classes / axis_dim;
-
     if (num_remain == 1 &&
         phi::backends::cpu::MayIUse(phi::backends::cpu::avx)) {
       const T* out_data = y->data<T>();
@@ -419,7 +421,9 @@ class SoftmaxGradFunctor<DeviceContext, T, enable_if_CPU<DeviceContext>> {
         out_grad += num_classes;
         in_grad += num_classes;
       }
-    } else {
+    } else
+#endif
+    {
       SoftmaxGradEigen<DeviceContext, T>()(
           context, axis_dim, y, y_grad, x_grad);
     }
