diff --git a/pyproject.toml b/pyproject.toml
index c50be8e8a6..cae72d9d63 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -28,7 +28,7 @@ dependencies = [
     "wikipedia==1.4.0",
     "qdrant-client==1.9.2",
     "weaviate-client>=4.10.2,<5.0.0",
-    "faiss-cpu==1.9.0.post1",
+    # "faiss-cpu==1.9.0.post1",
     "types-cachetools>=5.5.0.20240820,<6.0.0",
     "pymongo==4.10.1",
     "supabase>=2.6.0,<3.0.0",
