diff --git a/pyproject.toml b/pyproject.toml
index 75d59fe..b109ba6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [build-system]
 requires = [
     "setuptools>=54.0",
-    "setuptools_scm[toml]>=5.0",
+    "setuptools_scm[toml]>=5.0,<9.0",
     "wheel>=0.36.2",
     "Cython>=3.0.10",
     "tomli",
diff --git a/src/pymssql/_mssql.pyx b/src/pymssql/_mssql.pyx
index eed61dc..28d856d 100644
--- a/src/pymssql/_mssql.pyx
+++ b/src/pymssql/_mssql.pyx
@@ -668,7 +668,7 @@ cdef class MSSQLConnection:
             if encryption in TDS_ENCRYPTION_LEVEL:
                 DBSETLENCRYPT(login, TDS_ENCRYPTION_LEVEL[encryption])
             else:
-                raise ValueError(f"'encryption' option should be {TDS_ENCRYPTION_LEVEL.keys())} or None.")
+                raise ValueError(f"'encryption' option should be {list(TDS_ENCRYPTION_LEVEL.keys())} or None.")
 
         cdef bytes charset_bytes
         cdef char *_charset
@@ -855,7 +855,7 @@ cdef class MSSQLConnection:
             return int(<int>(<DBINT *>data)[0])
 
         elif dbtype == SQLINT8:
-            return long(<PY_LONG_LONG>(<PY_LONG_LONG *>data)[0])
+            return int(<PY_LONG_LONG>(<PY_LONG_LONG *>data)[0])
 
         elif dbtype == SQLFLT4:
             return float(<float>(<DBREAL *>data)[0])
@@ -1048,7 +1048,7 @@ cdef class MSSQLConnection:
             return 0
 
         if dbtype[0] in (SQLMONEY, SQLMONEY4, SQLNUMERIC, SQLDECIMAL):
-            if type(value) in (int, long, bytes):
+            if type(value) in (int, bytes):
                 value = decimal.Decimal(value)
 
             if type(value) not in (decimal.Decimal, float):
@@ -1992,7 +1992,7 @@ cdef _quote_simple_value(value, use_datetime2=False, charset='utf8'):
     if isinstance(value, float):
         return repr(value).encode(charset)
 
-    if isinstance(value, (int, long, decimal.Decimal)):
+    if isinstance(value, (int, decimal.Decimal)):
         return str(value).encode(charset)
 
     if isinstance(value, uuid.UUID):
