diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index e5611bd99ad..4d7ac88bfa1 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -171,6 +171,8 @@ if (project != rootProject) {
     distribution project(':distribution:archives:linux-tar')
     distribution project(':distribution:archives:oss-linux-tar')
     distribution project(':distribution:archives:oss-linux-aarch64-tar')
+    distribution project(':distribution:archives:linux-ppc64le-tar')
+    distribution project(':distribution:archives:oss-linux-ppc64le-tar')
 
     integTestRuntimeOnly(project(":libs:elasticsearch-core"))
   }
diff --git a/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java b/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
index c9a7e1eeb8e..99ef077c29c 100644
--- a/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/Architecture.java
@@ -11,7 +11,8 @@ package org.elasticsearch.gradle;
 public enum Architecture {
 
     X64,
-    AARCH64;
+    AARCH64,
+    PPC64LE;
 
     public static Architecture current() {
         final String architecture = System.getProperty("os.arch", "");
@@ -21,6 +22,8 @@ public enum Architecture {
                 return X64;
             case "aarch64":
                 return AARCH64;
+            case "ppc64le":
+                return PPC64LE;
             default:
                 throw new IllegalArgumentException("can not determine architecture from [" + architecture + "]");
         }
diff --git a/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java b/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
index c9d2f8a0f70..0948f83af65 100644
--- a/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java
@@ -24,7 +24,7 @@ import java.util.regex.Pattern;
 
 public class Jdk implements Buildable, Iterable<File> {
 
-    private static final List<String> ALLOWED_ARCHITECTURES = Collections.unmodifiableList(Arrays.asList("aarch64", "x64"));
+    private static final List<String> ALLOWED_ARCHITECTURES = Collections.unmodifiableList(Arrays.asList("aarch64", "x64", "ppc64le"));
     private static final List<String> ALLOWED_VENDORS = Collections.unmodifiableList(Arrays.asList("adoptopenjdk", "openjdk"));
     private static final List<String> ALLOWED_PLATFORMS = Collections.unmodifiableList(Arrays.asList("darwin", "linux", "windows", "mac"));
     private static final Pattern VERSION_PATTERN = Pattern.compile("(\\d+)(\\.\\d+\\.\\d+)?\\+(\\d+(?:\\.\\d+)?)(@([a-f0-9]{32}))?");
diff --git a/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java b/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
index b1065715442..f7ab1e7b2ef 100644
--- a/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
+++ b/buildSrc/src/test/java/org/elasticsearch/gradle/JdkDownloadPluginTests.java
@@ -84,7 +84,7 @@ public class JdkDownloadPluginTests extends GradleUnitTestCase {
             "11.0.2+33",
             "linux",
             "unknown",
-            "unknown architecture [unknown] for jdk [testjdk], must be one of [aarch64, x64]"
+            "unknown architecture [unknown] for jdk [testjdk], must be one of [aarch64, x64, ppc64le]"
         );
     }
 
diff --git a/distribution/archives/build.gradle b/distribution/archives/build.gradle
index 49ab50b1718..19d4d7e8b11 100644
--- a/distribution/archives/build.gradle
+++ b/distribution/archives/build.gradle
@@ -139,6 +139,13 @@ distribution_archives {
     }
   }
 
+  linuxPpc64leTar {
+    archiveClassifier = 'linux-ppc64le'
+    content {
+      archiveFiles(modulesFiles(false, 'linux-ppc64le'), 'tar', 'linux', 'ppc64le', false, true)
+    }
+  }
+
   ossLinuxAarch64Tar {
     archiveClassifier = 'linux-aarch64'
     content {
@@ -153,6 +160,13 @@ distribution_archives {
     }
   }
 
+  ossLinuxPpc64leTar {
+    archiveClassifier = 'linux-ppc64le'
+    content {
+      archiveFiles(modulesFiles(true, 'linux-ppc64le'), 'tar', 'linux', 'ppc64le', true, true)
+    }
+  }
+
   noJdkLinuxTar {
     archiveClassifier = 'no-jdk-linux-x86_64'
     content {
diff --git a/distribution/build.gradle b/distribution/build.gradle
index 3e905c3e880..2b27034e8c2 100644
--- a/distribution/build.gradle
+++ b/distribution/build.gradle
@@ -293,7 +293,7 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
   // Setup all required JDKs
   project.jdks {
     ['darwin', 'windows', 'linux'].each { platform ->
-      (platform == 'linux' ? ['x64', 'aarch64'] : ['x64']).each { architecture ->
+      (platform == 'linux' ? ['x64', 'aarch64', 'ppc64le'] : ['x64']).each { architecture ->
         "bundled_${platform}_${architecture}" {
           it.platform = platform
           it.version = VersionProperties.getBundledJdk(platform)
@@ -372,7 +372,7 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
         } else {
           buildModules = buildDefaultModulesTaskProvider
         }
-        List excludePlatforms = ['linux-x86_64', 'linux-aarch64', 'windows-x86_64', 'darwin-x86_64']
+        List excludePlatforms = ['linux-x86_64', 'linux-aarch64', 'windows-x86_64', 'darwin-x86_64', 'linux-ppc64le']
         if (platform != null) {
           excludePlatforms.remove(excludePlatforms.indexOf(platform))
         } else {
diff --git a/distribution/docker/build.gradle b/distribution/docker/build.gradle
index fbf231246ab..dc277749a06 100644
--- a/distribution/docker/build.gradle
+++ b/distribution/docker/build.gradle
@@ -23,6 +23,8 @@ configurations {
   aarch64OssDockerSource
   ossDockerSource
   transformLog4jJar
+  ppc64leDockerSource
+  ppc64leOssDockerSource
 }
 
 dependencies {
@@ -31,6 +33,8 @@ dependencies {
   aarch64OssDockerSource project(path: ":distribution:archives:oss-linux-aarch64-tar", configuration:"default")
   ossDockerSource project(path: ":distribution:archives:oss-linux-tar", configuration:"default")
   transformLog4jJar project(path: ":distribution:docker:transform-log4j-config", configuration: "default")
+  ppc64leDockerSource project(path: ":distribution:archives:linux-ppc64le-tar")
+  ppc64leOssDockerSource project(path: ":distribution:archives:oss-linux-ppc64le-tar")
 }
 
 ext.expansions = { Architecture architecture, boolean oss, DockerBase base, boolean local ->
@@ -40,6 +44,8 @@ ext.expansions = { Architecture architecture, boolean oss, DockerBase base, bool
       classifier = "linux-aarch64"
     } else if (architecture == Architecture.X64) {
       classifier = "linux-x86_64"
+    } else if (architecture == Architecture.PPC64LE) {
+      classifier = "linux-ppc64le"
     } else {
       throw new IllegalArgumentException("Unsupported architecture [" + architecture + "]")
     }
@@ -107,7 +113,7 @@ class SquashNewlinesFilter extends FilterReader {
 
 private static String buildPath(Architecture architecture, boolean oss, DockerBase base) {
   return 'build/' +
-    (architecture == Architecture.AARCH64 ? 'aarch64-' : '') +
+    (architecture == Architecture.AARCH64 ? 'aarch64-' : (architecture == Architecture.PPC64LE ? 'ppc64le-' : '')) +
     (oss ? 'oss-' : '') +
     (base == DockerBase.UBI ? 'ubi-' : '') +
     (base == DockerBase.UBI ? 'ubi-' : (base == DockerBase.IRON_BANK ? 'ironbank-' : '')) +
@@ -116,7 +122,7 @@ private static String buildPath(Architecture architecture, boolean oss, DockerBa
 
 private static String taskName(String prefix, Architecture architecture, boolean oss, DockerBase base, String suffix) {
   return prefix +
-    (architecture == Architecture.AARCH64 ? 'Aarch64' : '') +
+    (architecture == Architecture.AARCH64 ? 'Aarch64' : (architecture == Architecture.PPC64LE ? 'Ppc64le' : '')) +
     (oss ? 'Oss' : '') +
     (base == DockerBase.UBI ? 'Ubi' : (base == DockerBase.IRON_BANK ? 'IronBank' : '')) +
     suffix
@@ -197,6 +203,12 @@ void addCopyDockerContextTask(Architecture architecture, boolean oss, DockerBase
       } else {
         from configurations.aarch64DockerSource
       }
+    } else if (architecture == Architecture.PPC64LE) {
+      if (oss) {
+        from configurations.ppc64leOssDockerSource
+      } else {
+        from configurations.ppc64leDockerSource
+      }
     } else {
       if (oss) {
         from configurations.ossDockerSource
@@ -232,7 +244,7 @@ tasks.register("copyNodeKeyMaterial", Sync) {
 elasticsearch_distributions {
   Architecture.values().each { eachArchitecture ->
     Flavor.values().each { distroFlavor ->
-      "docker_$distroFlavor${ eachArchitecture == Architecture.AARCH64 ? '_aarch64' : '' }" {
+      "docker_$distroFlavor${ eachArchitecture == Architecture.AARCH64 ? '_aarch64' : (eachArchitecture == Architecture.PPC64LE ? '_ppc64le' : '') }" {
         architecture = eachArchitecture
         flavor = distroFlavor
         type = 'docker'
@@ -353,13 +365,13 @@ subprojects { Project subProject ->
   if (subProject.name.endsWith('-export')) {
     apply plugin: 'distribution'
 
-    final Architecture architecture = subProject.name.contains('aarch64-') ? Architecture.AARCH64 : Architecture.X64
+    final Architecture architecture = subProject.name.contains('aarch64-') ? Architecture.AARCH64 :  subProject.name.contains('ppc64le-') ? Architecture.PPC64LE : Architecture.X64
     final boolean oss = subProject.name.contains('oss-')
     // We can ignore Iron Bank at the moment as we don't
     // build those images ourselves.
     final DockerBase base = subProject.name.contains('ubi-') ? DockerBase.UBI : DockerBase.CENTOS
 
-    final String arch = architecture == Architecture.AARCH64 ? '-aarch64' : ''
+    final String arch = architecture == Architecture.AARCH64 ? '-aarch64' : (architecture == Architecture.PPC64LE ? '-ppc64le' : '')
     final String suffix = oss ? '-oss' : base == DockerBase.UBI ? '-ubi8' : ''
     final String extension = base == DockerBase.UBI ? 'ubi.tar' : 'docker.tar'
     final String artifactName = "elasticsearch${arch}${suffix}_test"
diff --git a/distribution/docker/src/docker/Dockerfile b/distribution/docker/src/docker/Dockerfile
index 472a92a3032..3d70da65666 100644
--- a/distribution/docker/src/docker/Dockerfile
+++ b/distribution/docker/src/docker/Dockerfile
@@ -50,6 +50,7 @@ RUN set -eux ; \\
     case "\$(arch)" in \\
         aarch64) tini_bin='tini-arm64' ;; \\
         x86_64)  tini_bin='tini-amd64' ;; \\
+        ppc64le) tini_bin='tini-ppc64le' ;; \\
         *) echo >&2 ; echo >&2 "Unsupported architecture \$(arch)" ; echo >&2 ; exit 1 ;; \\
     esac ; \\
     curl --retry 10 -S -L -O https://github.com/krallin/tini/releases/download/v0.19.0/\${tini_bin} ; \\
diff --git a/distribution/docker/src/docker/config/elasticsearch.yml b/distribution/docker/src/docker/config/elasticsearch.yml
index 50b154702b9..a58d9d46c23 100644
--- a/distribution/docker/src/docker/config/elasticsearch.yml
+++ b/distribution/docker/src/docker/config/elasticsearch.yml
@@ -1,2 +1,6 @@
 cluster.name: "docker-cluster"
 network.host: 0.0.0.0
+pack.ml.enabled: false
+discovery.seed_hosts: []
+bootstrap.system_call_filter: false
+discovery.type: single-node
\ No newline at end of file
diff --git a/distribution/packages/build.gradle b/distribution/packages/build.gradle
index f34a363cde1..8bd6cd1c0df 100644
--- a/distribution/packages/build.gradle
+++ b/distribution/packages/build.gradle
@@ -99,6 +99,8 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
     if (type == 'deb') {
       if (architecture == 'x64') {
         arch('amd64')
+      } else if (architecture == 'ppc64le') {
+        arch('ppc64le')
       } else {
         assert architecture == 'aarch64' : architecture
         arch('arm64')
@@ -107,6 +109,8 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
       assert type == 'rpm' : type
       if (architecture == 'x64') {
         arch('X86_64')
+      } else if (architecture == 'ppc64le') {
+        arch('ppc64le')
       } else {
         assert architecture == 'aarch64' : architecture
         arch('aarch64')
@@ -114,7 +118,7 @@ Closure commonPackageConfig(String type, boolean oss, boolean jdk, String archit
     }
     // Follow elasticsearch's file naming convention
     String jdkString = jdk ? "" : "no-jdk-"
-    String prefix = "${architecture == 'aarch64' ? 'aarch64-' : ''}${oss ? 'oss-' : ''}${jdk ? '' : 'no-jdk-'}${type}"
+    String prefix = "${architecture == 'aarch64' ? 'aarch64-' : (architecture == 'ppc64le' ? 'ppc64le-' : '')}${oss ? 'oss-' : ''}${jdk ? '' : 'no-jdk-'}${type}"
     destinationDirectory = file("${prefix}/build/distributions")
 
     // SystemPackagingTask overrides default archive task convention mappings, but doesn't provide a setter so we have to override the convention mapping itself
@@ -366,6 +370,22 @@ tasks.register('buildOssNoJdkDeb', Deb) {
   configure(commonDebConfig(true, false, 'x64'))
 }
 
+tasks.register('buildPpc64leDeb', Deb) {
+  configure(commonDebConfig(true, true, 'ppc64le'))
+}
+
+tasks.register('buildPpc64leOssDeb', Deb) {
+  configure(commonDebConfig(true, true, 'ppc64le'))
+}
+
+tasks.register('buildPpc64leRpm', Rpm) {
+  configure(commonRpmConfig(false, true, 'ppc64le'))
+}
+
+tasks.register('buildPpc64leOssRpm', Rpm) {
+  configure(commonRpmConfig(true, true, 'ppc64le'))
+}
+
 Closure commonRpmConfig(boolean oss, boolean jdk, String architecture) {
   return {
     configure(commonPackageConfig('rpm', oss, jdk, architecture))
diff --git a/buildSrc/version.properties b/buildSrc/version.properties
index 18d9cdb5955..0e36795a8b6 100644
--- a/buildSrc/version.properties
+++ b/buildSrc/version.properties
@@ -14,7 +14,7 @@ snakeyaml         = 1.26
 icu4j             = 62.1
 supercsv          = 2.4.0
 # when updating log4j, please update also docs/java-api/index.asciidoc
-log4j             = 2.11.1
+log4j             = 2.12.2
 slf4j             = 1.6.2

 # when updating the JNA version, also update the version in buildSrc/build.gradle
diff --git a/x-pack/plugin/watcher/build.gradle b/x-pack/plugin/watcher/build.gradle
index b3aca397f53..3f0a1ef8a8c 100644
--- a/x-pack/plugin/watcher/build.gradle
+++ b/x-pack/plugin/watcher/build.gradle
@@ -34,7 +34,7 @@ dependencies {
   testImplementation project(xpackModule('ilm'))

   // watcher deps
-  api 'com.googlecode.owasp-java-html-sanitizer:owasp-java-html-sanitizer:20191001.1'
+  api 'com.googlecode.owasp-java-html-sanitizer:owasp-java-html-sanitizer:20211018.1'
   runtimeOnly 'com.google.guava:guava:27.1-jre' // needed by watcher for the html sanitizer
   runtimeOnly 'com.google.guava:failureaccess:1.0.1'
   api 'com.sun.mail:jakarta.mail:1.6.4'

diff --git a/settings.gradle b/settings.gradle
index eba6a72456e..d1dbb8957ce 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -32,6 +32,8 @@ List projects = [
   'distribution:archives:linux-tar',
   'distribution:archives:oss-no-jdk-linux-tar',
   'distribution:archives:no-jdk-linux-tar',
+  'distribution:archives:oss-linux-ppc64le-tar',
+  'distribution:archives:linux-ppc64le-tar',
   'distribution:docker',
   'distribution:docker:docker-aarch64-build-context',
   'distribution:docker:docker-aarch64-export',
@@ -46,6 +48,10 @@ List projects = [
   'distribution:docker:ubi-docker-aarch64-export',
   'distribution:docker:ubi-docker-build-context',
   'distribution:docker:ubi-docker-export',
+  'distribution:docker:docker-ppc64le-build-context',
+  'distribution:docker:docker-ppc64le-export',
+  'distribution:docker:oss-docker-ppc64le-build-context',
+  'distribution:docker:oss-docker-ppc64le-export',
   'distribution:packages:aarch64-oss-deb',
   'distribution:packages:oss-deb',
   'distribution:packages:aarch64-deb',
@@ -58,6 +64,9 @@ List projects = [
   'distribution:packages:rpm',
   'distribution:packages:oss-no-jdk-rpm',
   'distribution:packages:no-jdk-rpm',
+  'distribution:packages:ppc64le-oss-deb',
+  'distribution:packages:ppc64le-deb',
+  'distribution:packages:ppc64le-rpm',
   'distribution:bwc:bugfix',
   'distribution:bwc:maintenance',
   'distribution:bwc:minor',