diff --git a/python/setup.py b/python/setup.py
index edf8580..0405eb8 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -69,6 +69,7 @@ _tbb = Extension("tbb._api", ["tbb/api.i"],
                        os.path.join(tbb_root, 'lib', 'intel64', 'gcc4.8'),  # for Linux
                        os.path.join(tbb_root, 'lib'),                       # for MacOS
                        os.path.join(tbb_root, 'lib', 'intel64', 'vc_mt'),   # for Windows
+                       os.path.join(tbb_root, 'lib64'),
                      ] if not use_compiler_tbb else [],
         language    ='c++',
         )
@@ -89,7 +90,7 @@ setup(  name        ="TBB",
         author      ="Intel Corporation",
         author_email="inteltbbdevelopers@intel.com",
         license     ="Dual license: Apache or Proprietary",
-        version     ="0.2",
+        version     ="2023.1.0",
         classifiers =[
             'Development Status :: 4 - Beta',
             'Environment :: Console',
