diff --git a/cmake/libraries.cmake b/cmake/libraries.cmake
index 96e299dd..9afa009e 100644
--- a/cmake/libraries.cmake
+++ b/cmake/libraries.cmake
@@ -4,7 +4,7 @@ set(FLB_PATH_LIB_RBTREE   "lib/rbtree")
 set(FLB_PATH_LIB_MSGPACK  "lib/msgpack-c")
 set(FLB_PATH_LIB_AVRO     "lib/avro")
 set(FLB_PATH_LIB_CHUNKIO  "lib/chunkio")
-set(FLB_PATH_LIB_LUAJIT   "lib/luajit-2.1.0-1e66d0f")
+set(FLB_PATH_LIB_LUAJIT   "lib/luajit2")
 set(FLB_PATH_LIB_MONKEY   "lib/monkey")
 set(FLB_PATH_LIB_JSMN     "lib/jsmn")
 set(FLB_PATH_LIB_MBEDTLS  "lib/mbedtls-2.24.0")
