diff --git a/CMakeLists.txt b/CMakeLists.txt
index 340896f86..81d6cb3aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -316,6 +316,10 @@ if (FLB_SYSTEM_MACOS)
   include(cmake/macos-setup.cmake)
 endif()

+if (FLB_SYSTEM_LINUX)
+  include(cmake/ppc64le.cmake)
+endif ()
+
 # Build for SystemZ - s390x arch
 if (FLB_SYSTEM_LINUX)
   include(cmake/s390x.cmake)
diff --git a/cmake/libraries.cmake b/cmake/libraries.cmake
index 4bc01c7d6..d8b22a51f 100644
--- a/cmake/libraries.cmake
+++ b/cmake/libraries.cmake
@@ -11,7 +11,7 @@ set(FLB_PATH_LIB_MSGPACK     "lib/msgpack-c")
 set(FLB_PATH_LIB_NGHTTP2     "lib/nghttp2")
 set(FLB_PATH_LIB_AVRO        "lib/avro")
 set(FLB_PATH_LIB_CHUNKIO     "lib/chunkio")
-set(FLB_PATH_LIB_LUAJIT      "lib/luajit-04dca791")
+set(FLB_PATH_LIB_LUAJIT      "lib/luajit2")
 set(FLB_PATH_LIB_MONKEY      "lib/monkey")
 set(FLB_PATH_LIB_JSMN        "lib/jsmn")
 set(FLB_PATH_LIB_SQLITE      "lib/sqlite-amalgamation-3450200")
diff --git a/cmake/luajit.cmake b/cmake/luajit.cmake
index b4dfca2b2..cc183693a 100644
--- a/cmake/luajit.cmake
+++ b/cmake/luajit.cmake
@@ -1,10 +1,71 @@
-# luajit cmake
-option(LUAJIT_DIR "Path of LuaJIT 2.1 source dir" ON)
-option(LUAJIT_SETUP_INCLUDE_DIR "Setup include dir if parent is present" OFF)
+# This file provides 'libluajit' target for both UNIX and Windows.
+#
+# To enable LuaJIT, include this file and link the build target:
+#
+#    include(cmake/luajit.cmake)
+#    target_link_libraries(fluent-bit libluajit)
+
+add_library(libluajit STATIC IMPORTED GLOBAL)
+
+# Global Settings
+set(LUAJIT_SRC ${CMAKE_CURRENT_SOURCE_DIR}/${FLB_PATH_LIB_LUAJIT})
+set(LUAJIT_DEST ${CMAKE_CURRENT_BINARY_DIR})
+
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(CFLAGS "${CFLAGS} -isysroot ${CMAKE_OSX_SYSROOT} -fno-stack-check")
+
+  if(DEFINED ENV{MACOSX_DEPLOYMENT_TARGET})
+    # Refer MACOSX_DEPLOYMENT_TARGET if available.
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=$ENV{MACOSX_DEPLOYMENT_TARGET}")
+  else()
+    # Calculate MACOSX_DEPLOYMENT_TARGET from sw_vers command result.
+    execute_process(COMMAND sw_vers -productVersion OUTPUT_VARIABLE MACOSX_DEPLOYMENT_TARGET)
+    string(REGEX REPLACE "\n$" "" MACOSX_DEPLOYMENT_TARGET ${MACOSX_DEPLOYMENT_TARGET})
+    string(REPLACE "." ";" DEPLOYMENT_VERSION_LIST ${MACOSX_DEPLOYMENT_TARGET})
+    list(GET DEPLOYMENT_VERSION_LIST 0 MACOSX_DEPLOYMENT_TARGET_MAJOR)
+    list(GET DEPLOYMENT_VERSION_LIST 1 MACOSX_DEPLOYMENT_TARGET_MINOR)
+    set(DEPLOYMENT_TARGET "MACOSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET_MAJOR}.${MACOSX_DEPLOYMENT_TARGET_MINOR}")
+  endif()
+else()
+  set(DEPLOYMENT_TARGET "")
+endif()
+
+# luajit (UNIX)
+# =============
+ExternalProject_Add(luajit
+  BUILD_IN_SOURCE TRUE
+  EXCLUDE_FROM_ALL TRUE
+  SOURCE_DIR ${LUAJIT_SRC}
+  CONFIGURE_COMMAND ./configure
+  BUILD_COMMAND $(MAKE) CC=${CMAKE_C_COMPILER} ${DEPLOYMENT_TARGET} CFLAGS=${CFLAGS} BUILDMODE=static "XCFLAGS=-fPIC"
+  INSTALL_COMMAND cp src/libluajit.a "${LUAJIT_DEST}/lib/libluajit.a")
+
+# luajit (Windows)
+# ================
+if (MSVC)
+  ExternalProject_Add(luajit-windows
+    BUILD_IN_SOURCE TRUE
+    EXCLUDE_FROM_ALL TRUE
+    SOURCE_DIR ${LUAJIT_SRC}/src
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ./msvcbuild.bat static
+    INSTALL_COMMAND ${CMAKE_COMMAND} -E copy lua51.lib "${LUAJIT_DEST}/lib/libluajit.lib")
+endif()
+
+# Hook the buld definition to 'libluajit' target
+if(MSVC)
+  add_dependencies(libluajit luajit-windows)
+  set(LUAJIT_STATIC_LIB "${LUAJIT_DEST}/lib/libluajit.lib")
+else()
+  add_dependencies(libluajit luajit)
+  set(LUAJIT_STATIC_LIB "${LUAJIT_DEST}/lib/libluajit.a")
+endif()
+
+set_target_properties(libluajit PROPERTIES IMPORTED_LOCATION "${LUAJIT_STATIC_LIB}")
+include_directories("${LUAJIT_DEST}/include/")
 set(LUAJIT_DIR ${FLB_PATH_ROOT_SOURCE}/${FLB_PATH_LIB_LUAJIT})
 include_directories(
   ${LUAJIT_DIR}/src
   ${CMAKE_CURRENT_BINARY_DIR}/lib/luajit-cmake
 )
-add_subdirectory("lib/luajit-cmake")
 set(LUAJIT_LIBRARIES "libluajit")
diff --git a/lib/monkey/deps/flb_libco/libco.c b/lib/monkey/deps/flb_libco/libco.c
index e0101d238..a66b5b311 100644
--- a/lib/monkey/deps/flb_libco/libco.c
+++ b/lib/monkey/deps/flb_libco/libco.c
@@ -16,7 +16,9 @@
     #include "arm.c"
   #elif defined(__aarch64__)
     #include "aarch64.c"
-  #elif defined(_ARCH_PPC)
+  #elif defined(__powerpc64__) && defined(_CALL_ELF) && _CALL_ELF == 2
+    #include "ppc64le.c"
+  #elif defined(_ARCH_PPC) && !defined(__LITTLE_ENDIAN__)
     #include "ppc.c"
   #elif defined(_WIN32)
     #include "fiber.c"
