diff --git a/conanfile.py b/conanfile.py
index 50bbaed..af1c2b8 100644
--- a/conanfile.py
+++ b/conanfile.py
@@ -27,6 +27,8 @@ class MilvusLiteConan(ConanFile):
         "openssl/1.1.1t",
         "libcurl/7.86.0",
         "grpc/1.50.1",
+        "opentelemetry-proto/1.3.2",
+        "opentelemetry-cpp/1.14.2",
         "prometheus-cpp/1.1.0",
         "re2/20230301",
         "simdjson/3.7.0",
@@ -41,7 +43,8 @@ class MilvusLiteConan(ConanFile):
         "libevent/2.1.12#4fd19d10d3bed63b3a8952c923454bc0",
     )

-    generators = {"cmake", "cmake_find_package"}
+    #generators = {"cmake", "cmake_find_package"}
+    generators = {"cmake", "cmake_find_package","CMakeDeps", "CMakeToolchain", "txt"}

     default_options = {
         "glog:with_gflags": True,
@@ -81,6 +84,7 @@ class MilvusLiteConan(ConanFile):
             self.options["onetbb"].tbbbind = False

     def requirements(self):
+        self.requires("abseil/20230802.1", override=True)
         if self.settings.os not in ["Macos", "Android"]:
             self.requires("libunwind/1.7.2")
         if self.settings.os == "Android":
@@ -93,3 +97,4 @@ class MilvusLiteConan(ConanFile):
     def build(self):
         target = "11.0"
         self.run("export MACOSX_DEPLOYMENT_TARGET={}".format(target))
+
diff --git a/thirdparty/milvus/internal/core/thirdparty/tantivy/tokenizer.h b/thirdparty/milvus/internal/core/thirdparty/tantivy/tokenizer.h
index 2870b01e2e..75a4d390e0 100644
--- a/thirdparty/milvus/internal/core/thirdparty/tantivy/tokenizer.h
+++ b/thirdparty/milvus/internal/core/thirdparty/tantivy/tokenizer.h
@@ -1,5 +1,15 @@
 #pragma once
-
+#ifndef AssertInfo
+#  include <stdexcept>
+#  define AssertInfo(cond, fmt_str, ...)               \
+     do {                                              \
+       if (!(cond))                                     \
+         throw std::runtime_error(                      \
+            /* simple formatting */                     \
+            fmt_str                                     \
+         );                                             \
+     } while (0)
+#endif
 #include "tantivy-binding.h"
 #include "rust-binding.h"
 #include "rust-hashmap.h"
