From ed336dfd3f847eef7f4790542d8fada6e295715c Mon Sep 17 00:00:00 2001
From: Srighakollapu-Sai-Srivatsa <Srighakollapu.Sai.Srivatsa@ibm.com>
Date: Mon, 28 Apr 2025 10:22:24 +0000
Subject: [PATCH] set-Protobuf-env-paths

---
 tools/setup_helpers/extension.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/setup_helpers/extension.py b/tools/setup_helpers/extension.py
index 2415bbae..59ea3787 100644
--- a/tools/setup_helpers/extension.py
+++ b/tools/setup_helpers/extension.py
@@ -140,6 +140,11 @@ class CMakeBuild(build_ext):
             f"-DUSE_CUDA:BOOL={'ON' if _USE_CUDA else 'OFF'}",
             f"-DUSE_OPENMP:BOOL={'ON' if _USE_OPENMP else 'OFF'}",
             f"-DUSE_FFMPEG:BOOL={'ON' if _USE_FFMPEG else 'OFF'}",
+            f"-DProtobuf_INCLUDE_DIR={os.getenv('Protobuf_INCLUDE_DIR')}",
+            f"-DProtobuf_LIBRARIES={os.getenv('Protobuf_LIBRARIES')}",
+            f"-DProtobuf_LIBRARY={os.getenv('Protobuf_LIBRARY')}",
+            f"-DProtobuf_LITE_LIBRARY={os.getenv('Protobuf_LITE_LIBRARY')}",
+            f"-DProtobuf_PROTOC_EXECUTABLE={os.getenv('Protobuf_PROTOC_EXECUTABLE')}",
         ]
         build_args = ["--target", "install"]
         # Pass CUDA architecture to cmake
--
2.43.5
