diff --git BUILD.boost BUILD.boost
index 104c726..d7d1863 100644
--- BUILD.boost
+++ BUILD.boost
@@ -57,6 +57,15 @@ config_setting(
     visibility = ["//visibility:public"],
 )
 
+config_setting(
+    name = "linux_ppc",
+    constraint_values = [
+        "@bazel_tools//platforms:linux",
+        "@bazel_tools//platforms:ppc",
+    ],
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "osx_x86_64",
     constraint_values = [
@@ -101,6 +110,11 @@ BOOST_CTX_ASM_SOURCES = select({
         "libs/context/src/asm/make_x86_64_sysv_elf_gas.S",
         "libs/context/src/asm/ontop_x86_64_sysv_elf_gas.S",
     ],
+    ":linux_ppc": [
+        "libs/context/src/asm/jump_ppc64_sysv_elf_gas.S",
+        "libs/context/src/asm/make_ppc64_sysv_elf_gas.S",
+        "libs/context/src/asm/ontop_ppc64_sysv_elf_gas.S",
+    ],
     ":osx_x86_64": [
         "libs/context/src/asm/jump_x86_64_sysv_macho_gas.S",
         "libs/context/src/asm/make_x86_64_sysv_macho_gas.S",
@@ -1602,6 +1616,7 @@ boost_library(
     defines = select({
         ":linux_arm": [],
         ":linux_x86_64": [],
+        ":linux_ppc": [],
         ":osx_x86_64": [],
         ":windows_x86_64": [
             "BOOST_ALL_NO_LIB",
