--- scipy/stats/stats.py        2022-03-23 09:46:07.309292660 +0000
+++ /scipy/scipy/stats/stats.py 2022-03-23 04:49:51.486636214 +0000
@@ -170,6 +170,7 @@

 import numpy as np
 from numpy import array, asarray, ma
+from numpy.lib import NumpyVersion

 from scipy.spatial.distance import cdist
 from scipy.ndimage import measurements
@@ -2915,9 +2916,16 @@
     if np.isnan(rng).any():
         raise ValueError("range must not contain NaNs")

+
     rng = sorted(rng)
-    pct = percentile_func(x, rng, axis=axis, interpolation=interpolation,
-                          keepdims=keepdims)
+
+    if NumpyVersion(np.__version__) >= '1.22.0':
+        pct = percentile_func(x, rng, axis=axis, method=interpolation,
+                              keepdims=keepdims)
+    else:
+        pct = percentile_func(x, rng, axis=axis, interpolation=interpolation,
+                              keepdims=keepdims)
+
     out = np.subtract(pct[1], pct[0])

     if scale != 1.0: