diff --git a/script/build-arm64.sh b/script/build-arm64.sh
index 807c7f7..4a5099d 100755
--- a/script/build-arm64.sh
+++ b/script/build-arm64.sh
@@ -6,27 +6,44 @@
 SOURCE=$1
 DESTINATION=$2
 BASEDIR=$3
-
+GIT_LFS_VERSION=2.4.0
 mkdir -p $DESTINATION

-docker run --rm --privileged multiarch/qemu-user-static:register --reset
-docker run -it \
---mount type=bind,source=$BASEDIR,target=$BASEDIR \
---mount type=bind,source=$DESTINATION,target=$DESTINATION \
--e "SOURCE=$SOURCE" \
--e "DESTINATION=$DESTINATION" \
--w=$BASEDIR \
---rm multiarch/debian-debootstrap:arm64-jessie sh $BASEDIR/script/build-arm64-git.sh
-cd - > /dev/null
+echo " -- Building git at $SOURCE to $DESTINATION"
+
+cd $SOURCE
+make clean
+DESTDIR="$DESTINATION" make strip install prefix=/ \
+           NO_PERL=1 \
+           NO_TCLTK=1 \
+           NO_GETTEXT=1 \
+           NO_OPENSSL=1 \
+           NO_INSTALL_HARDLINKS=1 \
+           CC='gcc' \
+           CFLAGS='-Wall -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -U_FORTIFY_SOURCE' \
+           LDFLAGS='-Wl,-Bsymbolic-functions -Wl,-z,relro'
+
+echo "-- Removing server-side programs"
+rm "$DESTINATION/bin/git-cvsserver"
+rm "$DESTINATION/bin/git-receive-pack"
+rm "$DESTINATION/bin/git-upload-archive"
+rm "$DESTINATION/bin/git-upload-pack"
+rm "$DESTINATION/bin/git-shell"
+
+echo "-- Removing unsupported features"
+rm "$DESTINATION/libexec/git-core/git-svn"
+rm "$DESTINATION/libexec/git-core/git-remote-testsvn"
+rm "$DESTINATION/libexec/git-core/git-p4"
+chmod 777 "$DESTINATION/libexec/git-core"

 echo "-- Building Git LFS"
 go get github.com/git-lfs/git-lfs
 GOPATH=`go env GOPATH`
 cd $GOPATH/src/github.com/git-lfs/git-lfs
 git checkout "v${GIT_LFS_VERSION}"
-$GOPATH/src/github.com/git-lfs/git-lfs/script/bootstrap -arch arm64 -os linux
+$GOPATH/src/github.com/git-lfs/git-lfs/script/bootstrap -arch ppc64le -os linux
 echo "-- Bundling Git LFS"
-GIT_LFS_FILE=$GOPATH/src/github.com/git-lfs/git-lfs/bin/releases/linux-arm64/git-lfs-$GIT_LFS_VERSION/git-lfs
+GIT_LFS_FILE=$GOPATH/src/github.com/git-lfs/git-lfs/bin/releases/linux-ppc64le/git-lfs-$GIT_LFS_VERSION/git-lfs
 SUBFOLDER="$DESTINATION/libexec/git-core"
cp $GIT_LFS_FILE $SUBFOLDER
