diff --git a/cmake/cpu_features.cmake b/cmake/cpu_features.cmake
index cfb0fadfabe..666566e3c8c 100644
--- a/cmake/cpu_features.cmake
+++ b/cmake/cpu_features.cmake
@@ -124,10 +124,10 @@ elseif (ARCH_PPC64LE)
     # Note that gcc and clang have support for x86 SSE2 intrinsics when building for PowerPC
     option (POWER9 "Build for Power 9 CPU and above" 0)
     if(POWER9)
-        set (COMPILER_FLAGS "${COMPILER_FLAGS} -maltivec -mcpu=power9 -D__SSE2__=1 -DNO_WARN_X86_INTRINSICS")
+        set (COMPILER_FLAGS "${COMPILER_FLAGS} -maltivec -mcpu=power9")
         list(APPEND RUSTFLAGS_CPU "-C" "target_feature=+power9-altivec")
     else ()
-        set (COMPILER_FLAGS "${COMPILER_FLAGS} -maltivec -mcpu=power8 -D__SSE2__=1 -DNO_WARN_X86_INTRINSICS")
+        set (COMPILER_FLAGS "${COMPILER_FLAGS} -maltivec -mcpu=power8")
         list(APPEND RUSTFLAGS_CPU "-C" "target_feature=+power8-altivec")
     endif ()
 
