diff --git a/build.gradle b/build.gradle
index ca13dade..dc4d0584 100644
--- a/build.gradle
+++ b/build.gradle
@@ -23,7 +23,7 @@ buildscript {
         }
         opensearch_no_snapshot = opensearch_version.replace("-SNAPSHOT","")
         common_utils_version = System.getProperty("common_utils.version", opensearch_build)
-        kotlin_version = '2.2.0'
+        kotlin_version = '2.2.20'
     }
 
     repositories {
diff --git a/gradle.properties b/gradle.properties
new file mode 100644
index 00000000..4d41702b
--- /dev/null
+++ b/gradle.properties
@@ -0,0 +1,4 @@
+org.gradle.native=false
+
+#Use plain console output (no rich formatting that requires native platform)
+org.gradle.console=plain

