diff --git a/ivy.xml b/ivy.xml
index 1836974..20f615a 100644
--- a/ivy.xml
+++ b/ivy.xml
@@ -53,7 +53,7 @@
     <dependency org="log4j" name="apache-log4j-extras" rev="1.0"/>
     <dependency org="net.freeutils" name="jcharset" rev="2.0"/>
     <dependency org="net.freeutils" name="jtnef" rev="1.9.0"/>
-    <dependency org="net.java.dev.jna" name="jna" rev="3.4.0"/>
+    <dependency org="net.java.dev.jna" name="jna" rev="4.4.0"/>
     <dependency org="net.oauth.core" name="oauth" rev="20100527"/>
     <dependency org="net.sf.ehcache" name="ehcache-core" rev="2.5.1"/>
     <dependency org="net.spy" name="spymemcached" rev="2.12.1" transitive="false"/>
@@ -124,7 +124,7 @@
     <dependency org="org.springframework" name="spring-context" rev="5.1.10.RELEASE"/>
     <dependency org="org.springframework" name="spring-core" rev="5.1.10.RELEASE"/>
     <dependency org="org.springframework" name="spring-expression" rev="5.1.10.RELEASE"/>
-    <dependency org="org.syslog4j" name="syslog4j" rev="0.9.46" transitive="false"/>
+    <dependency org="org.graylog2" name="syslog4j" rev="0.9.60" transitive="false"/>
     <dependency org="org.xerial" name="sqlite-jdbc" rev="3.7.15-M1"/>
     <dependency org="redis.clients" name="jedis" rev="2.9.0"/>
     <dependency org="wsdl4j" name="wsdl4j" rev="1.6.3"/>
diff --git a/pkg-builder.pl b/pkg-builder.pl
index 53ac064..9a08443 100755
--- a/pkg-builder.pl
+++ b/pkg-builder.pl
@@ -167,7 +167,7 @@ sub stage_zimbra_core_lib($)
         cpy_file("build/dist/jersey-server-1.11.jar",                               "$stage_base_dir/opt/zimbra/lib/jars/jersey-server-1.11.jar");
         cpy_file("build/dist/jersey-servlet-1.11.jar",                              "$stage_base_dir/opt/zimbra/lib/jars/jersey-servlet-1.11.jar");
         cpy_file("build/dist/jfreechart-1.0.15.jar",                                "$stage_base_dir/opt/zimbra/lib/jars/jfreechart-1.0.15.jar");
-        cpy_file("build/dist/jna-3.4.0.jar",                                        "$stage_base_dir/opt/zimbra/lib/jars/jna-3.4.0.jar");
+        cpy_file("build/dist/jna-4.4.0.jar",                                        "$stage_base_dir/opt/zimbra/lib/jars/jna-4.4.0.jar");
         cpy_file("build/dist/jsr181-api-1.0-MR1.jar",                               "$stage_base_dir/opt/zimbra/lib/jars/jsr181-api-1.0-MR1.jar");
         cpy_file("build/dist/jsr311-api-1.1.1.jar",                                 "$stage_base_dir/opt/zimbra/lib/jars/jsr311-api-1.1.1.jar");
         cpy_file("build/dist/junixsocket-common-2.0.4.jar",                         "$stage_base_dir/opt/zimbra/lib/jars/junixsocket-common-2.0.4.jar");
@@ -210,7 +210,7 @@ sub stage_zimbra_core_lib($)
         cpy_file("build/dist/stax-ex-1.7.7.jar",                                    "$stage_base_dir/opt/zimbra/lib/jars/stax-ex-1.7.7.jar");
         cpy_file("build/dist/stax2-api-3.1.1.jar",                                  "$stage_base_dir/opt/zimbra/lib/jars/stax2-api-3.1.1.jar");
         cpy_file("build/dist/streambuffer-2.2.6.jar",                               "$stage_base_dir/opt/zimbra/lib/jars/streambuffer-2.2.6.jar");
-        cpy_file("build/dist/syslog4j-0.9.46.jar",                                  "$stage_base_dir/opt/zimbra/lib/jars/syslog4j-0.9.46.jar");
+        cpy_file("build/dist/syslog4j-0.9.60.jar",                                  "$stage_base_dir/opt/zimbra/lib/jars/syslog4j-0.9.60.jar");
         cpy_file("build/dist/unboundid-ldapsdk-2.3.5.jar",                          "$stage_base_dir/opt/zimbra/lib/jars/unboundid-ldapsdk-2.3.5.jar");
         cpy_file("build/dist/woodstox-core-asl-4.2.0.jar",                          "$stage_base_dir/opt/zimbra/lib/jars/woodstox-core-asl-4.2.0.jar");
         cpy_file("build/dist/wsdl4j-1.6.3.jar",                                     "$stage_base_dir/opt/zimbra/lib/jars/wsdl4j-1.6.3.jar");
