diff --git a/discover/gpu.go b/discover/gpu.go
index 4a7d552d..103c96ff 100644
--- a/discover/gpu.go
+++ b/discover/gpu.go
@@ -9,6 +9,7 @@ import (
        "strings"
        "time"
 
+       "github.com/ollama/ollama/envconfig"
        "github.com/ollama/ollama/logutil"
        "github.com/ollama/ollama/ml"
 )
@@ -39,6 +40,10 @@ func GetSystemInfo() ml.SystemInfo {
                // Fall back to Go's num CPU
                threadCount = runtime.NumCPU()
        }
+        if userThreadCount := envconfig.NumThreads(); userThreadCount > 0 {
+               slog.Info("user override thread count", "override", userThreadCount, "default", threadCount)
+               threadCount = int(userThreadCount)
+       }
 
        return ml.SystemInfo{
                ThreadCount: threadCount,
diff --git a/envconfig/config.go b/envconfig/config.go
index f4bd4aa6..1e821b3d 100644
--- a/envconfig/config.go
+++ b/envconfig/config.go
@@ -273,6 +273,8 @@ func Uint(key string, defaultValue uint) func() uint {
 var (
        // NumParallel sets the number of parallel model requests. NumParallel can be configured via the OLLAMA_NUM_PARALLEL environment variable.
        NumParallel = Uint("OLLAMA_NUM_PARALLEL", 1)
+       // NumThreads overrides the default inference thread count.
+        NumThreads = Uint("OLLAMA_NUM_THREADS", 0)
        // MaxRunners sets the maximum number of loaded models. MaxRunners can be configured via the OLLAMA_MAX_LOADED_MODELS environment variable.
        MaxRunners = Uint("OLLAMA_MAX_LOADED_MODELS", 0)
        // MaxQueue sets the maximum number of queued requests. MaxQueue can be configured via the OLLAMA_MAX_QUEUE environment variable.
@@ -327,6 +329,7 @@ func AsMap() map[string]EnvVar {
                "OLLAMA_NOHISTORY":            {"OLLAMA_NOHISTORY", NoHistory(), "Do not preserve readline history"},
                "OLLAMA_NOPRUNE":              {"OLLAMA_NOPRUNE", NoPrune(), "Do not prune model blobs on startup"},
                "OLLAMA_NUM_PARALLEL":         {"OLLAMA_NUM_PARALLEL", NumParallel(), "Maximum number of parallel requests"},
+               "OLLAMA_NUM_THREADS":        {"OLLAMA_NUM_THREADS", NumThreads(), "Override the default cpu inference thread count"},
                "OLLAMA_ORIGINS":              {"OLLAMA_ORIGINS", AllowedOrigins(), "A comma separated list of allowed origins"},
                "OLLAMA_SCHED_SPREAD":         {"OLLAMA_SCHED_SPREAD", SchedSpread(), "Always schedule model across all GPUs"},
                "OLLAMA_MULTIUSER_CACHE":      {"OLLAMA_MULTIUSER_CACHE", MultiUserCache(), "Optimize prompt caching for multi-user scenarios"},
