diff --git a/maistra/ci/common.sh b/maistra/ci/common.sh
index ebc3a7c895..17249c24d0 100644
--- a/maistra/ci/common.sh
+++ b/maistra/ci/common.sh
@@ -21,6 +21,12 @@ COMMON_FLAGS="\
     --config=${ARCH} \
 "

+if [ "${ARCH}" = "ppc" ]; then
+  COMMON_FLAGS+=" --action_env=CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH \
+                 --action_env=Clang_DIR=$Clang_DIR \
+                 --action_env=LD_LIBRARY_PATH=$LD_LIBRARY_PATH"
+fi
+
 if [ -n "${BAZEL_REMOTE_CACHE}" ]; then
   COMMON_FLAGS+=" --remote_cache=${BAZEL_REMOTE_CACHE} "
 elif [ -n "${BAZEL_DISK_CACHE}" ]; then
